connection.go

```diff
@@ -1310,15 +1304,6 @@ func (conn *Connection) Do(req Request) *Future {
 			return fut
 		}
 	}
-	if req.Ctx() != nil {
-		select {
-		case <-req.Ctx().Done():
-			fut := NewFuture()
-			fut.SetError(fmt.Errorf("context is done"))
-			return fut
-		default:
-		}
-	}
```

Comment:
You haven't described, why this deletion?

Reply:
This is just a simplification of an optional check.

See the commit:

[code health: simplify context checks](https://github.com/tarantool/go-tarantool/pull/253/commits/091b93873ad1f99a55cdf357ce4213b4828b46b1)

We don't need to over-optimize the work with requests with contexts
that already done. It doesn't look like a work method. It is better
to simplify the code.

Part of https://github.com/tarantool/go-tarantool/issues/244

Reply:
But this way we avoided a putting a new request into the queue and starting a watchdog extra?

Reply:
We do not send a request in this case and don't start a watchdog at now too (because we have the same check after newFuture() and before putFuture()).

In fact, we were avoiding an unnecessary `newFuture` call with the check. But it doesn't look like too big problem: l assume that at a usual usecase a context does not done here.

Reply:
Ok, thank you for explanation.